File: tutorial2_topico/src/sub1.cpp
#include "ros/ros.h"
#include "tutorial2_topico/Variables.h"

void msgCallback(const tutorial2_topico::Variables::ConstPtr& v){
 ROS_INFO("resultado = %f", v->z);       // Shows the 'z' message
}

int main(int argc, char **argv){
ros::init(argc, argv, "sub1");
ros::NodeHandle nh;
ros::Subscriber ros_tutorial_sub = nh.subscribe("suma", 100, msgCallback);
// A function for calling a callback function, waiting for a message to be
// received, and executing a callback function when it is received
ros::spin();
return 0;
}

File: tutorial2_topico/src/sub1.py
#!/usr/bin/env python
import rospy #importar rospy
from tutorial2_topico.msg import Variables        #importar mensaje personalizado

def callback(data): #funcion a la que se llama cuando recibo un mensaje en el topic al que nos subscribimos
    rospy.loginfo("He recibido: %f, %f y %f", data.x, data.y, data.z) #escribe en el log el mensaje recibido
    rospy.loginfo("Resultado: %f ", data.z)


def subscriptor(): #funcion principal del nodo
    rospy.init_node('sub1', anonymous=True) #inicializa nodo con nombre subscriptor

    #se subscribe al topic suma con mensajes tipo Variables llamando a la funcion callback cuando se reciba mensaje
    rospy.Subscriber("suma", Variables, callback)
    rospy.spin()

if __name__ == '__main__':
    subscriptor()
